### PR TITLE
Cleaning up the OpBinScoreEvaluator class a little bit

### PR DESCRIPTION
**Pull request by tovbinm**
Removed some redundant ctor arguments and renames & idents.

### CODE REVIEW

File: core/src/main/scala/com/salesforce/op/evaluators/Evaluators.scala [hunk 1]
```diff
@@ -54,10 +54,11 @@ object Evaluators {
 
     /*
      * Brier Score for the prediction
+     *
+     * @param numOfBins number of bins to produce
      */
-    def brierScore(): OpBinScoreEvaluator =
-      new OpBinScoreEvaluator(
-        name = BinaryClassEvalMetrics.brierScore, isLargerBetter = true)
+    def brierScore(numOfBins: Int = 100): OpBinScoreEvaluator =
```
Comment (gokulsfdc) [inline core/src/main/scala/com/salesforce/op/evaluators/Evaluators.scala]: brierScore is independent of numBins.

Comment (tovbinm) [reply core/src/main/scala/com/salesforce/op/evaluators/Evaluators.scala]: Then perhaps we should not even expose it. Why is it 100 by default?

Comment (gokulsfdc) [reply core/src/main/scala/com/salesforce/op/evaluators/Evaluators.scala]: numBins are needed to compute other stats in OpBinScoreEvaluator like averageScore per bin.
As we are interested only in brierScore here, we don't need to pass numBins.

Comment (tovbinm) [reply core/src/main/scala/com/salesforce/op/evaluators/Evaluators.scala]: got it. removed.

**Pull request merged**